experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -257,3 +257,158 @@ function isomorphism(::Type{PermGroup}, W::WeylGroup; set_properties::Bool=true)
 
   return MapFromFunc(W, G, iso, isoinv)
 end
+
+@doc raw"""
+    map_word(w::WeylGroupElem, genimgs::Vector; genimgs_inv::Vector = genimgs, init = nothing)
+
+If `init` is `nothing` and `word(w) = [`$i_1$`, ..., `$i_n$`]`,
+then return the product $R_1 R_2 \cdots R_n$ with $R_j =$ `genimgs[`$i_j$`]`.
+Otherwise return the product $xR_1 R_2 \cdots R_n$ where $x =$ `init`.
+
+The length of `genimgs` must be equal to the rank of the parent of `w`.
+If `w` is the trivial element, then `init` is returned if it is different
+from `nothing`, and otherwise `one(genimgs[1])` is returned if `genimgs` is non-empty.
+If `w` is trivial, `init` is nothing and `genimgs` is empty, an error occurs.
+
+See also: [`map_word(::Union{FPGroupElem, SubFPGroupElem}, ::Vector)`](@ref),
+[`map_word(::Union{PcGroupElem, SubPcGroupElem}, ::Vector)`](@ref).
+Note that `map_word(::WeylGroupElem)` accepts but ignores the `genimgs_inv` keyword argument
+because the generators of a Weyl group are always self-inverse.
```

Comment:
```suggestion
Note that `map_word(::WeylGroupElem)` accepts the `genimgs_inv` keyword argument
for consistency with other `map_word` methods, but ignores it because the
generators of a Weyl group are always self-inverse.
```